addons/web_editor/static/lib/odoo-editor/src/commands/deleteBackward.js

```diff
@@ -110,7 +111,7 @@ HTMLElement.prototype.oDeleteBackward = function (offset, alreadyMoved = false,
         alreadyMoved = true;
         moveDest = endPos(leftNode);
     } else {
-        if (isUnremovable(this)) {
+        if (isUnremovable(this) || (closestElement(this,'TD') && this.textContent)) {
```

Comment:
Why? This seems out of place.

Reply:
@Zynto 
This handles the case when you create a heading inside a TD, write some text in it, place the cursor at the start, and hit the backspace key; then the heading tag is removed.

I have updated the code as it was missing the case where you have multiple empty heading previous siblings.

Reply:
It seems a bit hardcore to me to change that there and basically add a new contextual definition of "unremovable" just for this context. Was there no other place for this fix, that would be more appropriate? cc @dmo-odoo

Reply:
Continuing this discussion in the current code: https://github.com/odoo/odoo/pull/132778#discussion_r1507658909